### ISSUE TITLE
Panic when trying to delete an unmerged file

### ISSUE THREAD

**Issue by jesseduffield** [jesseduffield/lazygit#477]
**Describe the bug**
stacktrace is as follows:
```
2019/07/09 09:38:25 An error occurred! Please create an issue at https://github.com/jesseduffield/lazygit/issues

*errors.errorString error: path '<filename>' is unmerged

/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:152 (0x15d8314)
	sanitisedCommandOutput: return outputString, errors.New(outputString)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:64 (0x15d798d)
	(*OSCommand).RunCommandWithOutput: return sanitisedCommandOutput(cmd.CombinedOutput())
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:118 (0x15d2bc5)
	(*OSCommand).RunCommand: _, err := c.RunCommandWithOutput(command)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/commands/os.go:118 (0x15d2bad)
	(*OSCommand).RunCommand: _, err := c.RunCommandWithOutput(command)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/commands/git.go:482 (0x15d29b7)
	(*GitCommand).DiscardAllFileChanges: return c.DiscardUnstagedFileChanges(file)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/gui/files_panel.go:529 (0x1828bd8)
	(*Gui).handleCreateDiscardMenu.func1: return gui.GitCommand.DiscardAllFileChanges(file)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/gui/files_panel.go:557 (0x1828cf0)
	(*Gui).handleCreateDiscardMenu.func4: if err := options[index].handler(file); err != nil {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/gui/menu_panel.go:72 (0x1829df1)
	(*Gui).createMenu.func1: if err := handlePress(selectedLine); err != nil {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:748 (0x15ea9ef)
	(*Gui).execKeybinding: if err := kb.handler(g, v); err != nil {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:748 (0x15ea9dd)
	(*Gui).execKeybinding: if err := kb.handler(g, v); err != nil {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:695 (0x15ea81e)
	(*Gui).onKey: matched, err := g.execKeybindings(g.currentView, ev)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:458 (0x15e92e3)
	(*Gui).handleEvent: return g.onKey(ev)
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:456 (0x15e92bb)
	(*Gui).handleEvent: switch ev.Type {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/gui/gui.go:671 (0x1808117)
	(*Gui).Run: err = g.MainLoop()
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/gui/gui.go:680 (0x180827f)
	(*Gui).RunWithSubprocesses: if err := gui.Run(); err != nil {
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/pkg/app/app.go:161 (0x1835aab)
	(*App).Run: return app.Gui.RunWithSubprocesses()
/Users/jesseduffieldduffield/go/src/github.com/jesseduffield/lazygit/main.go:51 (0x1836438)
	main: err = app.Run()
/usr/local/Cellar/go/1.12/libexec/src/runtime/proc.go:200 (0x102eecc)
	main: fn()
/usr/local/Cellar/go/1.12/libexec/src/runtime/asm_amd64.s:1337 (0x10599d1)
	goexit: BYTE	$0x90	// NOP
```

**To Reproduce**
hit 'd' on an unmerged file showing the 'UU' status string.

**Expected behavior**
it just deletes the file or checks it out: whatever makes it go away.

**Desktop (please complete the following information):**
 - OS: OSX
 - Lazygit Version master
 - The last commit id 25c93c67

**Comment by giorgioprevitera** [comment]
@jesseduffield tried to fix this by changing the condition around the `git reset -- %s` command, see my PR.

**Comment by giorgioprevitera** [comment]
I guess this can be closed now that #479  has been merged

**Comment by jesseduffield** [comment]
will do!
